![New York City schoolbus](schoolbus.jpg)

Photo by [Jannis Lucas](https://unsplash.com/@jannis_lucas) on [Unsplash](https://unsplash.com).
<br>

Every year, American high school students take SATs, which are standardized tests intended to measure literacy, numeracy, and writing skills. There are three sections - reading, math, and writing, each with a **maximum score of 800 points**. These tests are extremely important for students and colleges, as they play a pivotal role in the admissions process.

Analyzing the performance of schools is important for a variety of stakeholders, including policy and education professionals, researchers, government, and even parents considering which school their children should attend. 

You have been provided with a dataset called `schools.csv`, which is previewed below.

You have been tasked with answering three key questions about New York City (NYC) public school SAT performance.

In [57]:
# Re-run this cell 
import pandas as pd

# Read in the data
schools = pd.read_csv("schools.csv")

# Preview the data
schools.head()


school_name  ... percent_tested
0  New Explorations into Science, Technology and ...  ...            NaN
1                               Essex Street Academy  ...           78.9
2                       Lower Manhattan Arts Academy  ...           65.1
3    High School for Dual Language and Asian Studies  ...           95.9
4      Henry Street School for International Studies  ...           59.7

[5 rows x 7 columns]

In [58]:
print(schools.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375 entries, 0 to 374
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   school_name      375 non-null    object 
 1   borough          375 non-null    object 
 2   building_code    375 non-null    object 
 3   average_math     375 non-null    int64  
 4   average_reading  375 non-null    int64  
 5   average_writing  375 non-null    int64  
 6   percent_tested   355 non-null    float64
dtypes: float64(1), int64(3), object(3)
memory usage: 20.6+ KB
None


## BEST SAT MATHS RESULT IN NYC SCHOOLS

In [59]:
def minimum_score(percentage, max_score):
    return (percentage / 100) * max_score
max_possible_score = 800
best_math_threshold = minimum_score(80, max_possible_score)
print(f"The best math results must be at least: {best_math_threshold}")

The best math results must be at least: 640.0


In [60]:
best_math_schools = schools[schools['average_math'] >= 640.0][['school_name', 'average_math']]
best_math_schools = best_math_schools.sort_values(by='average_math', ascending=False)
print(best_math_schools)

                                           school_name  average_math
88                              Stuyvesant High School           754
170                       Bronx High School of Science           714
93                 Staten Island Technical High School           711
365  Queens High School for the Sciences at York Co...           701
68   High School for Mathematics, Science, and Engi...           683
280                     Brooklyn Technical High School           682
333                        Townsend Harris High School           680
174  High School of American Studies at Lehman College           669
0    New Explorations into Science, Technology and ...           657
45                       Eleanor Roosevelt High School           641


The best math results are at least 80% of the maximum possible score of 800, the calculation shows that to become the best math, the result must be at least scored 640.0. 

Out of a total of 375 schools, 10 schools have demonstrated strong performance in their math programs with scores above the minimum threshold 640. Stuyvesant High School has the highest average math score of 754, with a margin of 40 points over the second-ranked Bronx High School of Science. The lowest score close to the minimum threshold is from Eleanor Roosevelt High School which achieved a score of 641.

## TOP 10 Performing Schools

In [61]:
schools["total_SAT"] = schools[["average_math", "average_writing", "average_reading"]].sum(axis=1) 
top_10_schools = schools[['school_name', 'total_SAT']].sort_values(by='total_SAT', ascending=False).head(10)
print(top_10_schools)

                                           school_name  total_SAT
88                              Stuyvesant High School       2144
170                       Bronx High School of Science       2041
93                 Staten Island Technical High School       2041
174  High School of American Studies at Lehman College       2013
333                        Townsend Harris High School       1981
365  Queens High School for the Sciences at York Co...       1947
5                       Bard High School Early College       1914
280                     Brooklyn Technical High School       1896
45                       Eleanor Roosevelt High School       1889
68   High School for Mathematics, Science, and Engi...       1889


The data presented highlights the top 10 performing high schools based on their combined SAT scores. This ranking provides insight into the academic excellence and competitiveness of these institutions and demonstrating their ability to prepare students for standardized testing and higher education. 

At the top of the list is Stuyvesant High School which is achieving an impressive combined SAT score of 2144, making it the highest performer. Following by Bronx High School of Science and Staten Island Technical High School with 2041 points. The High School of American Studies at Lehman College ranks fourth with 2013 points. Eleanor Roosevelt High School and High School for Mathematics, Science, and Engineering are in the 9th and 10th with 1889 points. The score range among the top 10 is relatively narrow, spanning from 2144 to 1889, indicating a highly competitive academic landscape. 

In summary, these leading schools demonstrate a deep dedication to academic success, offering students the tools and programs needed to perform well on standardized tests. With specialized courses, selective admissions, and challenging classes, they consistently achieve high SAT scores, solidifying their status as some of the top educational institutions in the nation.

## Identifying the Borough with the Largest Standard Deviation in Combined SAT Score Changes

In [62]:
borough_stats = schools.groupby("borough").agg(
    num_schools=("school_name", "size"),
    average_SAT=("total_SAT", "mean"),
    std_SAT=("total_SAT", "std")
).reset_index()
largest_std_dev = borough_stats.loc[borough_stats["std_SAT"].idxmax()]
largest_std_dev = pd.DataFrame([largest_std_dev]).round(2)
print(largest_std_dev)

     borough  num_schools  average_SAT  std_SAT
2  Manhattan           89      1340.13   230.29


The data indicates that Manhattan has the largest standard deviation of 230.29 in combined SAT score changes among all boroughs. A high standard deviation suggests significant variability in SAT scores across the 89 schools in the borough. This means that while some schools perform exceptionally well, others may have much lower scores and it's leading to a wide range of performance. The average SAT score in Manhattan is 1340.13 which provides a central benchmark, however the large spread implies that students’ performance is not uniform across different schools.